odoo/tools/misc.py

```diff
@@ -363,6 +363,27 @@ def add_sheet(self, name, cell_overwrite_ok=False):
     xlwt = None
 
 
+try:
+    import xlsxwriter
+
+    # add some sanitizations to respect the excel sheet name restrictions
+    # as the sheet name is often translatable, can not control the input
+    class PatchedXlsxWorkbook(xlsxwriter.Workbook):
+
+        def add_worksheet(self, name=None):
```

Comment:
I agree but Odoo use the 0.9.3 version in which there is no worksheet_class parameter.
(It caused a crash when I tested because I didn't check the version).
See here: https://github.com/jmcnamara/XlsxWriter/commit/cf4519b2b949fe4b0dfe08523742b1529016dd4c#diff-3c939515408e4baf3ccba336323b5644L160